### PR TITLE
Export zip tempfile

### PR DESCRIPTION
**Pull request by snazy**

### CODE REVIEW

File: versioned/transfer/src/main/java/org/projectnessie/versioned/transfer/files/ZipArchiveExporter.java [hunk 1]
```diff
@@ -45,13 +48,19 @@ public interface Builder {
 
   abstract Path outputFile();
 
+  @Value.Lazy
+  Path tempOutputFile() {
+    Path out = outputFile();
+    return out.resolveSibling("." + out.getFileName().toString() + ".tmp");
```
Comment (adutra) [inline versioned/transfer/src/main/java/org/projectnessie/versioned/transfer/files/ZipArchiveExporter.java:54]: Why not `Files.createTempFile()`?

**Review by adutra — COMMENTED** [review]
I'm not sure I follow the intent here: is to prevent failed exports to leave half-created zip files behind?

File: versioned/transfer/src/main/java/org/projectnessie/versioned/transfer/files/ZipArchiveExporter.java [hunk 2]
```diff
@@ -80,7 +89,14 @@ public OutputStream newFileOutput(@Nonnull @jakarta.annotation.Nonnull String fi
...
   @Override
   public void close() throws Exception {
-    zipOutput().close();
+    try {
+      zipOutput().close();
+    } finally {
+      deleteIfExists(outputFile());
+      if (isRegularFile(tempOutputFile())) {
+        move(tempOutputFile(), outputFile());
+      }
```
[the remark this answers is not in the document]
Comment (adutra) [reply versioned/transfer/src/main/java/org/projectnessie/versioned/transfer/files/ZipArchiveExporter.java:98]: Do you prefer to keep your version?

[the remark this answers is not in the document]
Comment (snazy) [reply versioned/transfer/src/main/java/org/projectnessie/versioned/transfer/files/ZipArchiveExporter.java:98]: Yes

**Pull request merged**